### PR TITLE
Fix bug #54: The navigation drawer should close first (if open) onBackPressed

### PR DESCRIPTION
**Pull request by XinyueZ**

### CODE REVIEW

**Comment by XinyueZ** [comment]
@tiembo I've resolved conflicts.

File: app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt [hunk 1]
```diff
@@ -88,4 +88,12 @@ class GardenActivity : AppCompatActivity() {
         // Pass any configuration change to the drawer toggle.
         drawerToggle.onConfigurationChanged(newConfig)
     }
+
+    override fun onBackPressed() {
+        with(binding.drawerLayout) {
+            val isDrawerOpen = isDrawerOpen(GravityCompat.START)
+            if (isDrawerOpen) closeDrawer(GravityCompat.START)
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: if/else should have braces

File: app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt [hunk 2]
```diff
@@ -88,4 +88,12 @@ class GardenActivity : AppCompatActivity() {
         // Pass any configuration change to the drawer toggle.
         drawerToggle.onConfigurationChanged(newConfig)
     }
+
+    override fun onBackPressed() {
+        with(binding.drawerLayout) {
+            val isDrawerOpen = isDrawerOpen(GravityCompat.START)
```
Comment (tiembo) [inline app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: let's inline `isDrawerOpen`

File: app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt (hunk 1 above)
Comment (XinyueZ) [reply app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: I have updated.

File: app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt (hunk 2 above)
Comment (XinyueZ) [reply app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: I have updated.

Comment (tiembo) [reply app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: Thanks.  Sorry that I wasn't clear - no need for an extension function; let's just use:
```
if (isDrawerOpen(GravityCompat.START)) {
  closeDrawer...
...
```

Comment (XinyueZ) [reply app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt]: @tiembo done, check it.

**Comment by tiembo** [comment]
Thanks!

**Pull request merged**